feat: show version staleness in error overlay (#44234)Co-authored-by: Sukka <isukkaw@gmail.com> Co-authored-by: Steven <steven@ceriously.com> Co-authored-by: Hannes Bornö <hannes.borno@vercel.com> Co-authored-by: Hannes Bornö <borno.hannes@gmail.com> Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com> (#44234)
vercel/next.js: errors/version-staleness.md

> # Version staleness

Reviewer: ```suggestion
# Version Staleness
```

#### Why This Error Occurred

> In the error overlay, a message was shown that the detected Next.js version was out of date or outdated.

Reviewer: > out of date or outdated

What's the difference between these two?

To get the newest features and bug fixes, it is recommended to stay up to date.

#### Possible Ways to Fix It

If you are testing out a canary release, upgrade Next.js with one of the following:

```sh
npm i next@canary
```

> or

Reviewer 2: ```suggestion
```

```sh
yarn add next@canary
```

> or

Reviewer 2: ```suggestion
```

```sh
pnpm add next@canary
```

If you are using a stable release, upgrade Next.js with one of the following:

```sh
npm i next@latest
```

> or

Reviewer 2: ```suggestion
```
We don't use or between the different install variations in examples so we probably shouldn't here either.

```sh
yarn add next@latest
```

or

```sh
pnpm add next@latest
```

> If you are coming from an older major version, check out our upgrade guides.

Reviewer 3: Is it possible to add a link to the upgrade guide on nextjs.org?

### Note

> If you want to report a bug on GitHub, it is required that you first check if `canary` has the issue, to make sure it hasn't already been fixed.

Reviewer 3: ```suggestion
If you want to report a bug on GitHub, you should upgrade to the latest canary release of Next.js has the issue first, to see if the bug has already been fixed.
```

> If you want to report a bug on GitHub, you should upgrade to the latest canary release of Next.js first, to see if the bug has already been fixed in canary.

Reviewer: ```suggestion
If you want to report a bug on GitHub, you should upgrade to the newest canary release of Next.js first, to see if the bug has already been fixed in canary.
```
